src/Voron/Impl/Scratch/ScratchBufferPool.cs

```diff
@@ -403,8 +403,12 @@ private void RemoveInactiveScratches(ScratchBufferItem except)
                 if (_scratchBuffers.TryRemove(scratchBufferItem.Number, out _) == false)
                     ThrowUnableToRemoveScratch(scratchBufferItem);
 
-                if (_recycleArea.Contains(scratchBufferItem) == false)
+                var isInRecycledArea = _recycleArea.Contains(scratchBufferItem);
+                if (cleanupRecycled || isInRecycledArea == false)
```

Comment:
I'm not following how this helps us, can you expand on this?

Reply:
I tried to use the existing function RemoveInactiveScratches. This solves the issue but doesn't clean the all the recycled scratches.
Fixed.